specs/src/specs/data_square_layout.md

```diff
@@ -25,28 +25,40 @@ First, we impose some ground rules:
-With the above constraint, we can compute subtree roots deterministically. For example, a blob of 128 shares and `SubtreeRootThreshold` (SRT) = 64, must start on a share index that is a multiple of 2 because 128/64 = 2. In this case, there will be a maximum of 1 share of padding between blobs (more on padding below). The maximum subtree width in shares will also be 2, meaning that there will be 2 shares under each subtree root.
+![](./figures/subtree_width.svg)
 
-The last piece of the puzzle is determining _which_ row the blob is placed at (or, more specifically, the starting location). This is needed to keep the block producer accountable. To this end, the block producer simply augments each fee-paying transaction with the starting locations of the blobs the transaction pays for.
+The `SubtreeRootThreshold` is an arbitrary versioned protocol constant that aims to put a soft limit on the number of subtree roots included in a blob inclusion proof, as described in [ADR013](../../../docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md). A higher `SubtreeRootThreshold` means less padding and more tightly packed squares but also means greater proof sizes.
+With the above constraint, we can compute subtree roots deterministically. For example, a blob of 172 shares and `SubtreeRootThreshold` (SRT) = 64, must start on a share index that is a multiple of 2 because 172/64 = 3. 3 rounded up to the nearest power of 2 is 4. In this case, there will be a maximum of 3 share of padding between blobs (more on padding below). The maximum subtree width in shares for the first mountain in the merkle range will also be 4 (The actual mountain range would be 43 subtree roots of 4 shares each).
 
-### Caveats
+## Padding
 
-The blob placement rules described above conflict with the first rule that shares must be ordered by namespace ID, as shares between two blobs that are not placed adjacent to each other do not have a natural namespace they belong to. This is resolved by requiring that such shares have a value of zero and a namespace ID equal to the preceding blob's. Since their value is known, they can be omitted from NMT proofs of all shares of a given namespace ID.
+Given these rules whereby blobs in their share format can't be directly appended one after the other, we use padding shares to fill the gaps. These are shares with all `0` bytes. Padding always comes after the blobs in the namespace. The padding at the end of the square (Tail Padding) is unique in that it belongs to it's own namespace. All other padding shares the namespace of the blob before it.
```

Comment:
[blocking] in addition to tail padding, there is also reserved padding: https://celestiaorg.github.io/celestia-app/specs/shares.html#reserved-padding-share

[optional]

```suggestion
Given these rules whereby blobs in their share format can't be directly appended one after the other, we use padding shares to fill the gaps. These are shares with a particular format (see [padding](./shares.md#padding)). Padding always comes after all the blobs in the namespace. The padding at the end of the reserved namespace and at the end of the square are special in that they belong to unique namespaces. All other padding shares use the namespace of the blob before it in the data square.
```

Reply:
> there is also reserved padding

Is reserved padding doing anything helpful? i.e. why can't it just be the padding belonging to the last namespace in the set of reserved namespaces?